Source/Turbo/Navigator/Routing/Handlers/SafariViewControllerRouteDecisionHandler.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import SafariServices
+
+/// Opens external URLs via an embedded `SafariViewController` so the user stays in-app.
+/// NOTE: This will silently fail for a URL that's not `http` or `https`.
+final class SafariViewControllerRouteDecisionHandler: RouteDecisionHandler {
+    let name: String = "safari"
+
+    func matches(location: URL,
+                 configuration: Navigator.Configuration) -> Bool {
```

Comment:
```suggestion
    func matches(location: URL,
                 configuration: Navigator.Configuration) -> Bool {
        guard location.scheme == "http" || location.scheme == "https" else { return false }
```

This would allow `http` and `https` links to open in the in-app browser but let `sms:` and `mailto:` links use `UIApplication.open()` by registering all three handlers. This might even be a better default as it would introduce fewer breaking changes.

```swift
Hotwire.registerRouteDecisionHandlers([
    AppNavigationRouteDecisionHandler(),
    SafariViewControllerRouteDecisionHandler(),
    BrowserRouteDecisionHandler()
])
```

Otherwise, we force developers to create their own decision handler for handling these types of schemes. It also makes `SafariViewControllerRouteDecisionHandler` and `BrowserRouteDecisionHandler` no longer mutually exclusive.

Reply:
I like the proposed change. In this case, `BrowserRouteDecisionHandler ` has become more generic.
I suggest we rename it to `SystemNavigationRouteDecisionHandler` or similar.

Reply:
I like this too 👍

Reply:
Yes! I like that a lot. I thought Browser vs. Safari was a little ambiguous but couldn't put my finger on why.